src/lib/comms/sol-oic-client.c

```diff
@@ -169,6 +169,21 @@ sol_oic_resource_ref(struct sol_oic_resource *r)
     return r;
 }
 
+static void
+clear_vector_list(struct sol_vector *vector, void *data)
+{
+    uint16_t i;
+    struct sol_str_slice *str;
+
+    if (data)
+        free(data);
+    else {
+        SOL_VECTOR_FOREACH_IDX(vector, str, i)
+            free((char *)str->data);
+    }
+    sol_vector_clear(vector);
```

Comment:
Yes and no.
We may have all strings in a single blob or multiple strings. If data is not NULL it is the first case and I can do a single free. If the opposite, I need to free all strings.
Well, I could have spited the blob string to have always the same case in free, but I didn't want to do it.
Can you see another way of checking if I have to free all strings or just the first one? If so, I'll do it.